app/assets/javascripts/osem-schedule.js

```diff
@@ -14,66 +17,99 @@ var Schedule = {
   },
   remove: function(element) {
     var e =  $("#" + element);
-    var event_schedule_id = e.attr("event_schedule_id");
-    if(event_schedule_id != null){
-      var my_url = url + '/' + event_schedule_id;
-      var success_callback = function(data) {
-        console.log(data);
-        e.attr("event_schedule_id", null);
-        e.appendTo($(".unscheduled-events"));
-        e.find(".schedule-event-delete-button").hide();
+    var event_id = e.attr("event_id");
+    if(e.attr("event_schedule_id")!= null){
+      events_to_remove[event_id] = {
+        event: e,
+        previous_parent: e.parent()
+      };
+    }
+    delete events_to_save[event_id];
+    var unscheduled = $(".unscheduled-events");
+    e.appendTo(unscheduled);
+    e.find(".schedule-event-delete-button").hide();
+  },
+  add: function (previous_parent, new_parent, event) {
+    var params = {
+      previous_parent: previous_parent,
+      new_parent: new_parent,
+      event: event
+    }
+    var event_id = event.attr("event_id");
+    events_to_save[event_id] = params;
+    delete events_to_remove[event_id];
+    event.appendTo(new_parent);
+    $("#event-" + event_id).find(".schedule-event-delete-button").show();
+  },
+  saveEvents: function () {
+    var errors = '';
+    for (var key in events_to_save){
+      var event = events_to_save[key]['event'];
+      var event_schedule_id = event.attr("event_schedule_id");
+      var new_parent = events_to_save[key]['new_parent'];
+      var my_url = url;
+      var type = 'POST';
+      var params = { event_schedule: {
+        room_id: new_parent.attr("room_id"),
+        start_time: (new_parent.attr("date") + ' ' + new_parent.attr("hour"))
+      }};
+      if(event_schedule_id != null){
+        type = 'PUT';
+        my_url += ('/' + event_schedule_id);
+      }
+      else{
+        params['event_schedule']['event_id'] = event.attr("event_id");
+        params['event_schedule']['schedule_id'] = schedule_id;
       }
-      var error_callback = function(data) {
+      var success_callback_save = function(data) {
+        console.log(data);
+        event.attr("event_schedule_id", data.event_schedule_id);
+        }
+      var error_callback_save = function(data) {
         console.log(data);
-        showError($.parseJSON(data.responseText).errors);
+        errors += $.parseJSON(data.responseText).errors;
+        event.appendTo(events_to_save[key]['previous_parent']);
```

Comment:
To unscheduled events or to its previous time and room if it was already scheduled.